pg/src/main/java/org/jdbi/v3/pg/SqlArrayArgumentFactory.java

```diff
@@ -0,0 +1,47 @@
+package org.jdbi.v3.pg;
+
+import java.util.Collections;
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import org.jdbi.v3.StatementContext;
+import org.jdbi.v3.tweak.Argument;
+import org.jdbi.v3.tweak.ArgumentFactory;
+
+public class SqlArrayArgumentFactory implements ArgumentFactory<Object[]> {
+
+    @Override
+    public boolean accepts(Class<?> expectedType, Object value, StatementContext ctx) {
+        return expectedType.isArray();
+    }
+
+    @Override
+    public Argument build(Class<?> expectedType, Object[] value, StatementContext ctx) {
+        return new ArrayArgument(guessSqlType(value), value);
+    }
```

Comment:
Sorry, you jumped on this PR before I was ready -- I branched off many months ago and am now trying to rebase on latest `jdbi3` and it's a bit of a mess at the moment :)